### ISSUE TITLE
Method ambiguity with new type that subtypes `StaticVector`

### ISSUE THREAD

**Issue by dpsanders** [JuliaArrays/StaticArrays.jl#123]
With the `julia-0.6` branch:
```jl
julia> using StaticArrays
INFO: Recompiling stale cache file /Users/dpsanders/.julia/lib/v0.6/StaticArrays.ji for module StaticArrays.

julia> Pkg.status("StaticArrays")
 - StaticArrays                  0.3.0+             julia-0.6

julia> immutable MyType{N,T} <: StaticVector{T}
           data::NTuple{N,T}
       end

julia> MyType(3, 4)
ERROR: MethodError: MyType{2,Int64}(::Tuple{Int64,Int64}) is ambiguous. Candidates:
  (::Type{MyType{N,T}})(data) where {N, T} in Main at REPL[3]:2
  (::Type{SA})(x::Tuple) where SA<:StaticArrays.StaticArray in StaticArrays at /Users/dpsanders/.julia/v0.6/StaticArrays/src/convert.jl:1
Stacktrace:
 [1] MyType(::Int64, ::Int64) at /Users/dpsanders/.julia/v0.6/StaticArrays/src/convert.jl:3

julia> versioninfo()
Julia Version 0.6.0-pre.alpha.208
Commit 003a14fd2* (2017-03-21 20:11 UTC)
Platform Info:
  OS: macOS (x86_64-apple-darwin16.0.0)
  CPU: Intel(R) Core(TM) i7-4870HQ CPU @ 2.50GHz
  WORD_SIZE: 64
  BLAS: libopenblas (USE64BITINT DYNAMIC_ARCH NO_AFFINITY Haswell)
  LAPACK: libopenblas64_
  LIBM: libopenlibm
  LLVM: libLLVM-3.9.1 (ORCJIT, haswell)
```

**Comment by andyferris** [comment]
Thanks for the report.

I'm guessing this comes down to new subtyping method?

**Comment by dpsanders** [comment]
Sorry, I'm not sure I understand the question. Isn't this the correct way to make a new type that subtypes StaticVector? 

This is a mwe educed from my real test case where I came across this problem in ValidatedNumerics.jl.

**Comment by andyferris** [comment]
The question was mostly to myself - on 0.6 dispatch matches and specificity has changed due to more accurate subtyping implemented by Jeff.

And no, I don't expect users to provide inner constructors to disambiguate this - this is a bug. It would be preferable to just remove that abstract type constructor.

I'll have a look into it.

**Comment by andyferris** [comment]
> Isn't this the correct way to make a new type that subtypes `StaticVector`?

I suspect you know this quite well, but you also need to define methods `Size(::Type{MyType{N,T}}) where {N,T} = Size(N)` and something for `getindex`.

**Comment by dpsanders** [comment]
Thanks. I do have those definitions in my package (thanks to your kind help), but even without them this method ambiguity arises.

**Comment by dpsanders** [comment]
Great work, by the way!

**Comment by andyferris** [comment]
Fixed on `julia-0.6`
